gtfs-validator/main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteUniqueNamesValidator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import com.google.auto.value.AutoValue;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.annotation.Inject;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.RouteUniqueNamesNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsRoute;
+import org.mobilitydata.gtfsvalidator.table.GtfsRouteTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsRouteType;
+
+/** Uses AutoValue to compare easier. */
+@AutoValue
+abstract class RouteIdentifier {
+  static RouteIdentifier create(
+      String longName, String shortName, GtfsRouteType routeType, String agencyId) {
+    return new AutoValue_RouteIdentifier(longName, shortName, routeType, agencyId);
+  }
+
+  abstract String longName();
+
+  abstract String shortName();
+
+  abstract GtfsRouteType routeType();
+
+  abstract String agencyId();
+}
+
+/**
+ * Validates whether the combination of names, type and agency ID for a route is unique.
+ *
+ * <p>Generated notices: {@link RouteUniqueNamesNotice}.
+ */
+@GtfsValidator
+public class RouteUniqueNamesValidator extends FileValidator {
+  @Inject GtfsRouteTableContainer routeTable;
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    final Map<RouteIdentifier, List<Object>> testedRoutes = new HashMap<>();
```

Comment:
Can we avoid the usage of List<Object>?
Using List<Object> reduces the ability of the compiler to check we have the expected set of details (especially when the used object is ArrayList.

this could be replaced by GtfsRoute of some AutoValue class containing routeId and csvRowNumber.

Reply:
Sure, I will do that.

Reply:
I have now changed it from the List of Object to another AutoValue to store routeId and csvRowNumber, as the value of the HashMap.